src/decycle.ts

```diff
@@ -1,36 +1,46 @@
 import { isPlainObject } from './isPlainObject';
 import { pathToPointer } from './pathToPointer';
 
-export const decycle = (obj: unknown, replacer?: (value: any) => any) => {
+export function decycle(obj: unknown, replacer?: (value: any) => any) {
   const objs = new WeakMap<object, string>();
-  return (function derez(value: any, path: string[]) {
-    // The new object or array
-    let curObj: any;
-
-    // If a replacer function was provided, then call it to get a replacement value.
-    if (replacer) value = replacer(value);
+  const objectsToDelete: object[] = []; // To keep track of objects to delete later
```

Comment:
rename objectsToDelete to objectsTobeDeleted

Reply:
changes are done